func_darwin.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2022 The Ebitengine Authors
+
+package purego
+
+import (
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/ebitengine/purego/internal/strings"
+)
+
+// RegisterLibFunc takes a pointer to a Go function representing the calling convention of the C function named name
+// found in the shared object provided by handle.
+// fptr will be set to a function that when called will call the C function given by name with the
+// parameters passed in the correct registers and stack.
+//
+// A panic is produced if the name symbol cannot be found in handle or if the type is not a function
+// pointer or if the function returns more than 1 value.
+//
+// These conversions describe how a Go type in the fptr will be used to call
+// the C function. It is important to note that there is no way to verify that fptr
+// matches the C function. This also holds true for struct types where the padding
+// needs to be ensured to match that of C; RegisterLibFunc does not verify this.
+//
+// Type Conversions (Go => C)
+//
+//	string => char*
+//	bool => _Bool
+//	uintptr => uintptr_t
+//	uint => uint32_t or uint64_t
+//	uint8 => uint8_t
+//	uint16 => uint16_t
+//	uint32 => uint32_t
+//	uint64 => uint64_t
+//	int => int32_t or int64_t
+//	int8 => int8_t
+//	int16 => int16_t
+//	int32 => int32_t
+//	int64 => int64_t
+//	float32 => float
+//	float64 => double
+//	struct => struct
+//	func => C function
+//	[]T, unsafe.Pointer, *T => void*
+//
+// There is a special case when the last argument of fptr is a variadic interface (or []interface}
```

Comment:
Just out of curiosity, would it be hard to support other types than `interface{}`?

Reply:
Do you have an example?

Also note that this does not support calling variadic functions like `int printf(const char* format, ...);` It's basically a cast to the desired C function. `(*(char*, int)())(printf)("%d", 3)` is not the same as `printf("%d, 3)` since the second uses va_args under the hood. The first is what is currently handled

Reply:
For example: https://pkg.go.dev/reflect#Append (I don't think this can be a C function but this is an example with a non-interface{} type)

Reply:
I now understand that the purpose of `RegisterFunc` is to have a Go function that is a counterpart of a C function. So, having another type than `interface{}` would not make sense.

By the way, in the comment, the order of types is `(Go type) => (C type)`, but is there an order here?

Reply:
Yes the Go function should have the same type as the C function being called.

There is no order except for how I thought of the type

Reply:
As for the Append example that would be a C function like `Append(Value s, Value *v)`. `...` in Go is just a slice and slices of any type are supported. Perhaps this example will clear things up about what `...interface{}` means in the context of `RegisterFunc`. This is only really useful for objc_msgSend variants since it is completely different from C's variadic which use a different calling convention then normally passing those arguments.

```go
var foo func(string, ...any)
purego.RegisterLibFunc(&foo, handle, "foo")
foo("hello", 123) // calls foo(string, int)
foo("hello", 1.342) // calls foo(string, double)
```
Neither of these calls are `foo(string, ...)`.

Reply:
I understood.

> Yes the Go function should have the same signature as the C function being called.

Maybe `<=>` would be better than `=>`?

Reply:
I like that since some types like slice cannot be returned from C to Go since there is no way to get its length

Reply:
OK so I'm fine with `=>`. Still LGTM